File: sorting/bubbleSort.java
class BubbleSort{
//Bubble Sort 
	public static int[] bubbleSort(int A[]) {
		int aux;
		long exchanges = 0;
		long comparisons = 0;
		
		for(int i = 1;i<=A.length;i++) {
			comparisons++;
			for(int k = 0;k<A.length-1;k++) {
				comparisons++;
				if(A[k]>A[k+1]) {
					comparisons++;
					aux = A[k];
					A[k] = A[k+1];
					A[k+1] = aux;
					exchanges =+ 3;
				}
			}
		}
		System.out.println("Exchanges: "+exchanges+" Comparisons: "+comparisons);
		return A;
		}
}

File: sorting/bubble_sort.cpp
// Implementation of Bubble Sort in C++ using array as data structure

#include <iostream>
using namespace std;

int main()
{ 
  int array[100], i, j, temp;
 
  cout<<"Input the number of array elements: ";
  cin>>n;
 
  cout<<"Input the array elements: ";
  for(i=0; i<n; i++) //Input elements of the array
    {                               
      cin>>array[i];
      cout<<" ";
    }
 
  for(i=0; i<n; i++)
    { 
      for(j=0; j<n-i-1; j++)
          {
            if(array[j] > array[j+1] 
                { 
                   temp = array[j+1];.  //Swaps the elements
                   array[j+1] = array[j];
                   array[j] = temp;
                }
          }
    }
  
  cout<<"The sorted array is: ";
  for(i=0; i<n; i++) 
    cout<<array[i]<<" "; //Displays the sorted array
  
  return 0;
}  
